### ISSUE TITLE
animation onEnd remove/destroy sugar

### ISSUE THREAD

**Issue by pvdz** [uxebu/bonsai#42]
Please discuss this with some people, but I think it's nice to have simple sugar to remove an element after an animation ends (especially when fading out). Something like

`bs.animate('3s', {opacity:0}, {onEnd:'remove'})` where `remove` is indeed just a string, which would be substituted by a function that removes the animated objects when the animation ends.

Same for `destroy`, since I guess those are the two most common things. Although there might be more...?

**Comment by padolsey** [comment]
cc @basecode @davidaurelio 

Personally, I feel this would be unnecessary.

**Comment by davidaurelio** [comment]
hm, `{onEnd: bind(subject.remove, subject)}` should do the trick.

I know, our `bind` is called “hitch” at the moment and has a bad parameter order, but we can fix that

**Comment by nonken** [comment]
If we had +1s in github ^^ +1 :) should have a little "tricks" section somewhere in the docs or so

**Comment by pvdz** [comment]
The main problem is the bloat of doing the same. And having to create a handle.

So right now I often do this:

`new Rect(10,10,50,50,3).attr({...}).addTo(stage).fadestroy('2s','3s');`

I dont have to save a handle for this object now so there's no chance of leaking it anywhere other than the stage itself. It's also very short and clean, imo. Opposed to say;

`var x = new Rect(10,10,50,50,3).attr({...}).addTo(stage).animate('2s',{opacity:0},{delay:'3s',onEnd:function(x){x.destroy(); x=null;});`

or even `this`, if that would be usable (wasn't last time i checked..)

`new Rect(10,10,50,50,3).attr({...}).addTo(stage).animate('2s',{opacity:0},{delay:'3s',onEnd:function(){this.destroy();});`

I use my `fadestroy` plugin a lot for temporary elements. I don't feel strong about adding it to bonsai by default, but I'm just saying that I use it a lot right now.

Alternatively, you could have string shortcuts for the third parameter. `'destroy'`, `'remove'`, or like a numbered parameter for delay. I think those would be really nice without affecting the "regular" (or future) api when using an object.

**Comment by basecode** [comment]
I agree that  `onEnd:'remove'` is a common use-case. Also `fadeIn` and `fadeOut`, but that's a different story.

Can't we put the necessary code in a plugin and see what works best? I guess we want to support a more general `onEnd:'keyword'`.
